### PR TITLE
Netapp driver poc for replication

### PR DESCRIPTION
**Pull request by PravinRanjan10**
The documents guide to setup netapp driver for replication with soda. This includes the installation and how to test guide.

You can find the youtube video link below:
https://www.youtube.com/watch?v=N-9b67tJHYg&list=PLSpuGyLJjJFQZcNnSWGgQ0L5tr6h0WVbG&index=2

**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind new feature
> /kind bug fix
> /kind cleanup
> /kind revert change
> /kind design
> /kind documentation
> /kind enhancement

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Test Report Added?**:
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:
> /kind TESTED
> /kind NOT-TESTED

**Test Report**:

**Special notes for your reviewer**:

### CODE REVIEW

File: NetappReplication/NetappReplicationDriverPoC.md [hunk 1]
```diff
@@ -0,0 +1,159 @@
+# Volume based replication with SODA Netapp driver PoC
```
Comment (kumarashit) [inline NetappReplication/NetappReplicationDriverPoC.md:1]:
```suggestion
# PoC for NetApp Volume replication using SODA Driver
```

File: NetappReplication/NetappReplicationDriverPoC.md [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+1. Clone all the three repo(api controller and dock):
+
+```
+git clone https://github.com/sodafoundation/api.git
+git clone https://github.com/sodafoundation/controller.git
+git clone https://github.com/sodafoundation/dock.git
+```
+
+2. Add the driver codes in dock repo:
+
```
Comment (kumarashit) [inline NetappReplication/NetappReplicationDriverPoC.md:29]: put inside code block

File: NetappReplication/NetappReplicationDriverPoC.md [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+ $GOPATH/dock/contrib/drivers/netapp/ontap/dock/contrib/drivers/netapp/ontap/replication.go
+
+ $GOPATH/dock/contrib/drivers/netapp/ontap/dock/contrib/drivers/netapp/ontap/cli.go
+
+
+ 3. Modify replication_driver.go file and add for netapp driver in dock repo.
+
+ vim contrib/drivers/replication_drivers.go
```
Comment (kumarashit) [inline NetappReplication/NetappReplicationDriverPoC.md:37]: put inside code block

File: NetappReplication/NetappReplicationDriverPoC.md [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+
+ a. add import path
+ ```
+ import (
+         "github.com/sodafoundation/dock/contrib/drivers/netapp/ontap"
+)
+```
+b. NetappDRBDDriverType drivertype
```
Comment (kumarashit) [inline NetappReplication/NetappReplicationDriverPoC.md:45]: Don't put DRBD. It will be confusing

File: NetappReplication/cli.go [hunk 5]
```diff
@@ -0,0 +1,118 @@
+// Copyright 2019 The OpenSDS Authors.
```
Comment (kumarashit) [inline NetappReplication/cli.go:1]: Any particular reason not to use the API for snap mirror?

File: NetappReplication/NetappReplicationDriverPoC.md [hunk 6]
```diff
@@ -0,0 +1,159 @@
...
+```
+```
+enabled_backends = lvm,nfs,netapp_ontap_san
+```
+
+8. Add the netapp_ontap_san.yml to /ete/opensds/driver directory
+
+ netapp_ontap_san.yml
+
+ ```
```
Comment (kumarashit) [inline NetappReplication/NetappReplicationDriverPoC.md:101]: Please mention clearly that this PoC is for replication of volumes from same aggregate on a SVM

File: NetappReplication/cli.go (hunk 5 above)
Comment (PravinRanjan10) [reply NetappReplication/cli.go:1]: Yes. currently vendors package of netapp also doesn't have API for snap mirror.

File: NetappReplication/NetappReplicationDriverPoC.md (hunk 6 above)
Comment (PravinRanjan10) [reply NetappReplication/NetappReplicationDriverPoC.md:101]: Done. Resolved all comments.

**Review by kumarashit — APPROVED** [review]
LGTM

**Review by skdwriting — APPROVED** [review]
LGTM

**Pull request merged**